src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-button-strategy.ts

```diff
@@ -74,6 +93,184 @@ export default class PaypalCommerceButtonStrategy implements CheckoutButtonStrat
         });
     }
 
+    private _transformContactToAddress(details: PayerDetails, address: CurrentShippingAddress) {
+        const contact = {
+            firstName: details.payer.name.given_name,
+            lastName: details.payer.name.surname,
+            email: details.payer.email_address,
+            address1: details.purchase_units[0].shipping.address.address_line_1,
+        };
+
+        return  {
+            ...address,
+            ...contact,
+        };
+    }
+
+    private async _onHostedMethodApprove(_data: ApproveDataOptions, actions: ApproveActions, cart: Cart) {
+        const orderPlacement = this._paymentMethod?.initializationData?.intent === 'capture'
+            ? await actions.order.capture()
+            : await actions.order.authorize();
+
+        if (this._currentShippingAddress) {
+            const shippingAddress = this._transformContactToAddress(orderPlacement, this._currentShippingAddress);
+            const lineItems = this._collectLineItems(cart.lineItems);
+            const consignmentPayload = [{
+                shippingAddress,
+                lineItems,
+            }];
+
+            try {
+               const state = await this._store.dispatch(this._consignmentActionCreator.createConsignments(consignmentPayload));
+               const consignments = state.consignments.getConsignments();
+
+               if (this._selectedShippingOptionId && consignments) {
+                   await this._store.dispatch(this._consignmentActionCreator.updateConsignment({...consignmentPayload[0], id: consignments[0].id}));
+                }
```

Comment:
It might be better to fix the issue with consignment in another way:

```suggestion
            const consignment = {
                shippingAddress,
                lineItems,
            };

            try {
               const state = await this._store.dispatch(this._consignmentActionCreator.createConsignments([consignment]));
               const consignments = state.consignments.getConsignments();

               if (this._selectedShippingOptionId && consignments) {
                   await this._store.dispatch(this._consignmentActionCreator.updateConsignment({...consignment, id: consignments[0].id}));
                }
```